### DISCUSSION TITLE
static cpu manager policy is not work as expected when I apply a helmchart that create a deployment

### DISCUSSION THREAD

**Discussion by x-coder-L** [k3s-io/k3s#8747] [Q&A]
**Environmental Info:**
k3s version v1.27.2+k3s1 (213d7ad4)
go version go1.20.4
rocky-linux8 kernel: 5.17.15-1.el8.x86_64

Node(s) CPU architecture, OS, and Version: 

Cluster Configuration:

**Describe the bug:**

I set cpu manager policy to static and I assign reserved-cpu: 0,5-25,30-50.It works successful when I apply a nginx pod with cpu request equal to limit and equal to 8.But it did't work as expected when I deleted this pod and then apply a helmchart with deployment which also with cpu request equal to limit and equal to 8.This is the cpu_manager_state when I don't apply nginx and helmchart.
![image](https://github.com/k3s-io/k3s/assets/62658324/b997177e-59e0-4a50-8db9-f970cb3cb5ed)
After I apply nginx pod the cpu_manager_state is:
![image](https://github.com/k3s-io/k3s/assets/62658324/04f203f0-5efe-43f9-8aaa-f61d1085d61f)
After I delete nginx pod and then apply helmchart:
![image](https://github.com/k3s-io/k3s/assets/62658324/39987a30-58dd-45f2-8629-69428cc0bffe)
it's not work when I apply helmchart to create this pod.
Here is my nginx pod yaml.
![image](https://github.com/k3s-io/k3s/assets/62658324/70c85bdc-15bd-405d-8a8b-921777bb23ad)
Here is some info about my deployment yaml in helmchart.
![image](https://github.com/k3s-io/k3s/assets/62658324/fac46430-8ed5-4579-9cfe-0f8c59ce8199)
and this deployment using hostNetwork.
**Steps To Reproduce:**

1. add   - "cpu-manager-policy=static" - "reserved-cpus=xx" to /etc/rancher/k3s/config.yaml
2. drain the node
3. systemctl stop k3s
4. delete /var/lib/kubelet/cpu_manager_state
5. systemctl start k3s
6. kubectl apply helmchart(which contained a deployment with cpu request equal to cpu limit and equal to a interge)
7. check cpu_manager_state

**Expected behavior:**

I hope the pod's cpuset will changed and my process can run in constant cpu.
**Actual behavior:**

**Additional context / logs:**

**Comment by brandond** [comment]
K3s does not have any special logic for handling cpu-manager configuration; whatever behavior you're seeing is the same as upstream. I suspect that there is a problem with your helm template; I would probably recommend comparing the rendered chart output (with `helm template`) where your resources do not work, vs your test pod where they do.

**Comment by x-coder-L** [comment]
> K3s does not have any special logic for handling cpu-manager configuration; whatever behavior you're seeing is the same as upstream. I suspect that there is a problem with your helm template; I would probably recommend comparing the rendered chart output (with `helm template`) where your resources do not work, vs your test pod where they do.

ok thakns. I will try to compare helm template vs my test pod.
